proj3_chess/chess/chess_pieces.py

```diff
@@ -0,0 +1,151 @@
+from board import Board
+from piece_properties import Player
+from abc import ABC, abstractmethod
+from coordinates import Move, Spot, MoveVector
+from typing import Iterable, Optional
+
+RIGHT = MoveVector(x=1, y=0)
+LEFT = MoveVector(x=-1, y=0)
+UP = MoveVector(x=0, y=1)
+DOWN = MoveVector(x=0, y=-1)
+UP_RIGHT = MoveVector(x=1, y=1)
+UP_LEFT = MoveVector(x=-1, y=1)
+DOWN_RIGHT = MoveVector(x=1, y=-1)
+DOWN_LEFT = MoveVector(x=-1, y=-1)
+
+
+class ChessPiece(ABC):
+    PIECE_NAME = ""
+
+    def __init__(self, player: Player) -> None:
+        self.player = player
+        self.direction = player.direction.value
+        self.symbol = player.symbol_transformation(self.PIECE_NAME)
+
+    def add_to_board(self, board: Board, spot: Spot) -> None:
```

Comment:
This is better than editing the board in the contructor, since at least we do it more explicitly now.
However we still keep a **two-way reference** - `ChessPiece` has a pointer to `Board` and vice versa. This is always a dangerous position to be in, since it gets complicated quick. It's hard to distinguish who is responsible for what.

### If you take one thing away from this project - this should be it. 
When you design your code, think about the responsibility of each class and make sure your design avoids these **two-way references** as best you can.